Automattic/wp-calypso
# Sharing: Refactor sharing buttons to use Redux instead of Flux Stores

In this PR I'm refactoring the Sharing Buttons Component to use Redux Actions instead of Flux Stores. This PR includes:

 - Using site-settings redux subtree
 - Using sharing-buttons redux subtree
 - Using post-types redux subtree
 - Dropping the sharing buttons list Flux Store
 - Refactor the components using the last coding standards (No mixins, no analytics lib, no notices lib)
 - Avoid using the `notices`middleware for SiteSettings notices, this would have the effect of triggering multiple notices for the same "save". So, I moved this behaviour to the component instead.

One nice to have enhancement (that could be done in another PR) is to avoid using the `site` Prop. This would require refactoring the underlying components.

**Testing instructions**

 - Open the `/sharing/buttons/$site`
 - Try to edit and save the sharing buttons settings
 - The component should continue working as expected
 - Try to edit/save the General Settings form as well, the success/error notices should show up as expected.
